sdk/core/src/wallet/metamask.ts

```diff
@@ -0,0 +1,97 @@
+import { WalletBase } from './base.js';
+import {
+  BrowserProvider,
+  Transaction,
+  hexlify,
+  type TransactionRequest,
+  type Eip1193Provider,
+} from 'ethers';
+
+declare global {
+  // Expose the injected provider placed on the window object by MetaMask.
+  // We type it as unknown because the exact shape is library-specific and we
+  // interact with it exclusively through ethers' BrowserProvider wrapper.
+  interface Window {
+    ethereum?: unknown;
+  }
+}
+
+/**
+ * Wallet integration for the MetaMask browser extension.
+ *
+ * It fulfils the minimal `WalletBase` contract (fetching the current address
+ * and signing arbitrary messages). MetaMask **cannot** sign a raw transaction
+ * without also broadcasting it, therefore {@link signTxn} intentionally throws
+ * and consumers should use {@link sendTransaction} instead.
+ */
+export class MetamaskWallet extends WalletBase {
+  private constructor(private readonly provider: BrowserProvider) {
+    super();
+  }
+
+  /**
+   * Request connection to MetaMask and create a new `MetamaskWallet`.
+   *
+   * Internally this triggers the extension UI via `eth_requestAccounts` which
+   * asks the user to authorise account access.
+   *
+   * @throws If no injected provider is found (MetaMask not installed).
+   */
+  public static async connect(): Promise<MetamaskWallet> {
+    if (typeof window.ethereum === 'undefined') {
```

Comment:
I'm thinking - do we really want the SDK to handle wallet-specific connetions? or should we just instead have clients provide the connected wallet (providing they respect `WalletBase`)? 
This way frontends can choose to support any wallet they want